### PR TITLE
feat: implement log level

### PR DESCRIPTION
**Pull request by shalev007**
- feat: add log types functions
- feat: use injected logger
- test: fix tests
- feat: rename logger
- refactor: remove LogUtil
- feat: replace old log request
- feat: add additional log data to every log
- feat: add logLevel to helm values
- feat: connect log level to helm values
- test: fix test file
- feat: replace print with logger
- refactor: use injected logger
- refactor: rename variable to logger

### CODE REVIEW

File: pkg/controllers/validationController.go [hunk 1]
```diff
@@ -73,18 +76,18 @@ func (c *ValidationController) Validate(w http.ResponseWriter, req *http.Request
...
-			internalLogger.LogError(utils.ParseErrorToString(panicErr))
+			c.logger.LogError(utils.ParseErrorToString(panicErr))
 			warningMessages = append(warningMessages, "Datree failed to validate the applied resource. Check the pod logs for more details.")
 			writer.WriteBody(services.ParseEvaluationResponseIntoAdmissionReview(admissionReviewReq.Request.UID, true, utils.ParseErrorToString(panicErr), warningMessages))
 		}
 	}()
 
-	internalLogger.LogIncoming(admissionReviewReq)
-	admissionReview, isSkipped := c.ValidationService.Validate(admissionReviewReq, &warningMessages, internalLogger)
+	c.logger.LogAdmissionRequest(admissionReviewReq, false, "incoming")
```
Comment (shmu3l) [inline pkg/controllers/validationController.go:85]: enum `"incoming"` ` "outgoing"`?

File: pkg/serviceState/serviceState.go [hunk 2]
```diff
@@ -32,9 +34,29 @@ type ServiceState struct {
 	verbose           string
 	bypassPermissions *BypassPermissions
 	enabledWarnings   string
+	LogLevel          zapcore.Level
 }
 
 func New() *ServiceState {
+	rawLogLEvels := os.Getenv(enums.LogLevel)
+	logLevel := zapcore.InfoLevel
+	switch rawLogLEvels {
```
Comment (shmu3l) [inline pkg/serviceState/serviceState.go:43]: think we should add it to docs/readme

File: pkg/controllers/validationController.go [hunk 3]
```diff
@@ -41,11 +43,12 @@ func NewValidationController(cliServiceClient *clients.CliClient, state *service
 	return &ValidationController{
 		ValidationService: validationService,
 		ErrorReporter:     errorReporter,
+		logger:            logger,
 	}
 }
 
 func (c *ValidationController) Validate(w http.ResponseWriter, req *http.Request) {
-	internalLogger := logger.New(uuid.NewString(), c.ErrorReporter)
+	c.logger.SetRequestId(uuid.NewString())
```
Comment (royhadad) [inline pkg/controllers/validationController.go:51]: BUG
this will override for parallel requests

File: pkg/logger/logger.go [hunk 4]
```diff
@@ -23,64 +23,64 @@ type Logger struct {
...
 	return Logger{
 		zapLogger:     zapLogger,
-		requestId:     requestId,
 		errorReporter: errorReporter,
 	}
 }
 
-func (l *Logger) LogError(message string) {
-	l.zapLogger.Error(message, zap.String("requestId", l.requestId))
+func (l *Logger) SetRequestId(requestId string) {
```
Comment (royhadad) [inline pkg/logger/logger.go:41]: remove this

File: pkg/logger/logger.go [hunk 5]
```diff
@@ -23,64 +23,64 @@ type Logger struct {
...
-	logger := New("", nil)
-	logger.LogInfo(msg)
+func (l *Logger) LogAndReportUnexpectedError(message string) {
+	l.LogError(message)
+	l.errorReporter.ReportUnexpectedError(errors.New(message))
 }
 
-func (l *Logger) logInfo(objectToLog any, requestDirection string) {
+func (l *Logger) LogAdmissionRequest(admissionReview *admission.AdmissionReview, isSkipped bool, direction string) {
+	if direction != "incoming" && direction != "outgoing" {
```
Comment (royhadad) [inline pkg/logger/logger.go:75]: This is not perfect but I think it's better
image

**Review by royhadad — APPROVED** [review]

**Pull request merged**